### ISSUE TITLE
Unable to improve accuracy with the existing cnn for text classification code for image classification

### ISSUE THREAD

**Issue by 7wik** [apache/incubator-mxnet#6368]
1.I tried to use the existing cnn for text classification code for image classification. I made appropriate changes  in the code.
2.The code is running without any errors.
`1.I` have avoided using the functions which were used for converting word2 vectors and all other word-vector conversion functions.
2.changed the forward propagation in cnn by using standard cnn model->output of each layer is fed as input to successive layers unlike the forward prop in cnn for text model 
`3.from __future__ import print_function

Model = namedtuple("Model", ['mod_exec', 'symbol', 'data', 'label', 'param_blocks'])
def product(tuple1):
	prod = 1
	for x in tuple1:
		prod = prod * x
	return prod 

def our_model(batch_size=100,filter_sizes=[3,3,3],filter_count=[8,16,32]):
	input_x = mx.sym.Variable('data',shape=(batch_size,784))
	input_y = mx.sym.Variable('softmax_label',shape=(batch_size,10))

	x_tensor=mx.sym.Reshape(data=input_x, target_shape=(batch_size, 1, 28, 28))

	current_input=x_tensor

	encoder = []
	shapes  = []
	activations= []

	for layer_i, n_output in enumerate(filter_count):
		W=mx.sym.Variable(name='weight'+str(layer_i),shape=(filter_count[layer_i],int((current_input.infer_shape()[1])[0][1]),
			filter_sizes[layer_i],filter_sizes[layer_i]))
		b=mx.sym.Variable('bias'+str(layer_i),shape=(filter_count[layer_i],))
		out = mx.sym.Convolution(data=current_input,weight=W,bias=b,num_filter=filter_count[layer_i],kernel=(filter_sizes[layer_i],filter_sizes[layer_i]),stride=(2,2))
		out = mx.sym.relu(data=out)
		current_input=out
		encoder.append(W)
		activations.append(out)
	#out = mx.sym.Dropout(data=out,p=0.5)
	t=product((out.infer_shape()[1])[0])
	out=mx.sym.Reshape(data=out, target_shape=(batch_size,t/batch_size))
	cls_weight = mx.sym.Variable('cls_weight',shape=(10,t/batch_size))
	cls_bias = mx.sym.Variable('cls_bias',shape=(10,))

	fc = mx.sym.FullyConnected(data=out, weight=cls_weight, bias=cls_bias, num_hidden=10)
	sm = mx.sym.SoftmaxOutput(data=fc, label=input_y, name='softmax')

	return sm

def setup_model(ctx, batch_size,initializer=mx.initializer.Xavier()):
	mod = our_model(batch_size=batch_size)
	arg_names = mod.list_arguments()
	input_shapes = {}
	input_shapes['data']=[batch_size,784]
	arg_shape, out_shape, aux_shape = mod.infer_shape(**input_shapes)
	arg_arrays = [mx.nd.zeros(s, ctx) for s in arg_shape]
	args_grad = {}
	for shape, name in zip(arg_shape, arg_names):
		if name in ['softmax_label','data']:
			continue
		args_grad[name] = mx.nd.zeros(shape, ctx)
	
	param_blocks = []
	arg_dict = dict(zip(arg_names, arg_arrays))
	for i, name in enumerate(arg_names):
		if name in ['softmax_label','data']:
			continue
		arg_dict[name][:]=np.random.uniform(-0.1, 0.1, arg_dict[name].shape)

		#arg_dict[name][:]=(mx.init.Xavier(shape=arg_dict[name].shape))#
		param_blocks.append( (i, arg_dict[name], args_grad[name], name) )
	
	mod_exec = mod.bind(ctx=ctx, args=arg_dict, args_grad=args_grad, grad_req='add')
	out_dict = dict(zip(mod.list_outputs(), mod_exec.outputs))
	data = mod_exec.arg_dict['data']
	label = mod_exec.arg_dict['softmax_label']

	return Model(mod_exec=mod_exec, symbol=mod, data=data, label=label, param_blocks=param_blocks)
def train_cnn(model, X_train_batch, y_train_batch, X_dev_batch, y_dev_batch, batch_size,optimizer='rmsprop', max_grad_norm=5.0, learning_rate=0.05, epoch=10):
	m = model
	opt = mx.optimizer.create(optimizer)
	opt.lr = learning_rate

	updater = mx.optimizer.get_updater(opt)
	for iteration in range(epoch):
		tic = time.time()
		num_correct = 0
		num_total = 0
		for begin in range( X_train_batch.shape[0] / batch_size):
			batchX = X_train_batch[begin*batch_size:(begin+1)*batch_size,:]
			batchY = y_train_batch[begin*batch_size:(begin+1)*batch_size]
			if batchX.shape[0] != batch_size:
				continue
			m.data[:] = batchX
			m.label[:] = batchY

			m.mod_exec.forward(is_train=True)
			#m.mod_exec.FeedForward(is_train=True)
			m.mod_exec.backward()

			num_correct += np.sum((mx.nd.equal(mx.nd.argmax(mx.nd.array(batchY),axis=1) ,mx.nd.argmax(m.mod_exec.outputs[0], axis=1))).asnumpy())

			num_total += len(batchY)
			norm = 0
			for idx, weight, grad, name in m.param_blocks:
				grad /= batch_size
				l2_norm = mx.nd.norm(grad).asscalar()
				norm += l2_norm * l2_norm
				norm = math.sqrt(norm)

			for idx, weight, grad, name in m.param_blocks:
				if norm > max_grad_norm:
					grad *= (max_grad_norm / norm)
				updater(idx, grad, weight)
				grad[:] = 0.0

		'''if iteration % 3 == 0 and iteration > 0:
			opt.lr *= 0.5
			print('reset learning rate to %g' % opt.lr,file=logs)'''
		toc = time.time()
		train_time = toc - tic
		train_acc = num_correct * 100 / float(num_total)
		num_correct = 0
		num_total = 0
		for begin in range(0, X_dev_batch.shape[0]/ batch_size):
			batchX = X_dev_batch[begin*batch_size:(begin+1)*batch_size]
			batchY = y_dev_batch[begin:begin+batch_size]

			if batchX.shape[0] != batch_size:
				continue

			m.data[:] = batchX
			m.mod_exec.forward(is_train=False)
			num_correct += np.sum((mx.nd.equal(mx.nd.argmax(mx.nd.array(batchY),axis=1) ,mx.nd.argmax(m.mod_exec.outputs[0], axis=1))).asnumpy())
			num_total += len(batchY)

		dev_acc = num_correct * 100 / float(num_total)
		print('Iter [%d] Train: Time: %.3fs, Training Accuracy: %.3f--- Dev Accuracy thus far: %.3f' % (iteration, train_time, train_acc, dev_acc), file=logs)
x_train=np.reshape(x_train,[-1,784])
x_test = np.reshape(x_test,[-1,784])

batch_size = 100
model = setup_model(mx.cpu(0), batch_size=100,initializer=mx.initializer.Xavier())
train_cnn(model, x_train, y_train, x_test, y_test, batch_size=100,optimizer='rmsprop', max_grad_norm=5.0, learning_rate=0.05, epoch=10)

`

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
